Classes/BITMetricsManager.m

```diff
@@ -74,13 +74,12 @@ - (void)startManager {
 
 - (void)setDisabled:(BOOL)disabled {
   if (_disabled == disabled) { return; }
-  
+    _disabled = disabled;
   if (disabled) {
     [self unregisterObservers];
   } else {
-    [self registerObservers];
+    [self startManager];
```

Comment:
No need to generate a new session UUID

Reply:
I agree that it makes sense to start a new session in case the feature was disabled and then enabled again.